timeserver/timeserver_test.go

```diff
@@ -0,0 +1,98 @@
+package timeserver
+
+import (
+	"net/http"
+	"testing"
+
+	"io/ioutil"
+	"net/http/httptest"
+)
+
+func BenchmarkTimer(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		TimeHandler := timer{timerProv{}}
+
+		ts := httptest.NewServer(http.HandlerFunc(TimeHandler.RealTime))
+
+		defer ts.Close()
+
+		resp, err := http.Get(ts.URL)
+		if err != nil {
+			panic(err)
+		}
+
+		defer resp.Body.Close()
+
+		_, err = ioutil.ReadAll(resp.Body)
+		if err != nil {
+			panic(err)
+		}
+
+	}
+
+}
+
+type fakeTimerProv struct{}
+
+func (ft fakeTimerProv) getCurrentTime() string {
+	return "2017-04-27 11:49:33.7154352 +0300 EEST"
+}
+
+func TestGetTimeOK(t *testing.T) {
+	t.Log("Start")
+	var expectedResult = "2017-04-27 11:49:33.7154352 +0300 EEST"
+
+	fakeTimeHandler := timer{fakeTimerProv{}}
+
+	ts := httptest.NewServer(http.HandlerFunc(fakeTimeHandler.RealTime))
```

Comment:
not necessary to launch `httptest.Server`, instead, use `httptest.ResponseRecorder` and `http.Request` with `handlerFunc`